src/sqlfluff/rules/references/RF01.py

```diff
@@ -91,30 +91,30 @@ def _eval(self, context: RuleContext) -> EvalResultType:
         ):
             return LintResult()
 
+        if FunctionalContext(context).parent_stack.any(sp.is_type(*_START_TYPES)):
+            return LintResult()
```

Comment:
My only issue with this is the mypy strict stuff. Between this and the below, we have 3 types of return, List[LintResult] (violations), LintResult, or None - could do with a bit of cleaning. Really we should only have one Falsy return type

Reply:
That's a good point - I've made them all lists, which I think is best in this case